src/integrationTest/java/org/opensearch/security/CrossClusterSearchTests.java

```diff
@@ -0,0 +1,428 @@
+		.dls(String.format("{\"match\":{\"%s\":\"%s\"}}", FIELD_GENRE, GENRE_JAZZ))
+		.on(SONG_INDEX_NAME);
+
+	private static final Role FLS_EXCLUDE_LYRICS_ROLE = new Role("fls_exclude_lyrics_role")
+		.indexPermissions("indices:data/read/search", "indices:data/read/get", "indices:admin/shards/search_shards")
+		.fls("~" + FIELD_LYRICS)
+		.on(SONG_INDEX_NAME);
+
+	private static final Role FLS_INCLUDE_TITLE_ROLE = new Role("fls_include_title_role")
+		.indexPermissions("indices:data/read/search", "indices:data/read/get", "indices:admin/shards/search_shards")
+		.fls(FIELD_TITLE)
+		.on(SONG_INDEX_NAME);
+
+	public static final String TYPE_ATTRIBUTE = "type";
+
+	private static final User ADMIN_USER = new User("admin").roles(ALL_ACCESS).attr(TYPE_ATTRIBUTE, "administrative");
+	private static final User LIMITED_USER = new User("limited_user").attr(TYPE_ATTRIBUTE, "personal");
+
+	private static final User FLS_INCLUDE_TITLE_USER = new User("fls_include_title_user");
+
+	private static final User FLS_EXCLUDE_LYRICS_USER = new User("fls_exclude_lyrics_user");
+
+	private static final User DLS_USER_ROCK = new User("dls-user-rock");
+
+	private static final User DLS_USER_JAZZ = new User("dls-user-jazz");
+
+	public static final String LIMITED_USER_INDEX_NAME = "user-" + LIMITED_USER.getName() + "-" + LIMITED_USER.getAttribute(TYPE_ATTRIBUTE);
+	public static final String ADMIN_USER_INDEX_NAME = "user-" + ADMIN_USER.getName() + "-" + ADMIN_USER.getAttribute(TYPE_ATTRIBUTE);
+
+	private static final TestCertificates TEST_CERTIFICATES = new TestCertificates();
+
+	private final boolean ccsMinimizeRoundtrips;
+
+	public static final String PLUGINS_SECURITY_RESTAPI_ROLES_ENABLED = "plugins.security.restapi.roles_enabled";
+	@ClassRule
+	public static final LocalCluster remoteCluster = new LocalCluster.Builder().certificates(TEST_CERTIFICATES)
+		.clusterManager(ClusterManager.SINGLENODE).anonymousAuth(false).clusterName(REMOTE_CLUSTER_NAME)
+		.authc(AUTHC_HTTPBASIC_INTERNAL)
+		.roles(LIMITED_ROLE, DLS_ROLE_ROCK, DLS_ROLE_JAZZ, FLS_EXCLUDE_LYRICS_ROLE, FLS_INCLUDE_TITLE_ROLE).users(ADMIN_USER)
+		.build();
+
+	@ClassRule
+	public static final LocalCluster cluster = new LocalCluster.Builder().certificates(TEST_CERTIFICATES)
+		.clusterManager(ClusterManager.SINGLE_REMOTE_CLIENT).anonymousAuth(false).clusterName("ccsLocal")
+		.nodeSettings(Map.of(PLUGINS_SECURITY_RESTAPI_ROLES_ENABLED, List.of("user_" + ADMIN_USER.getName()  +"__" + ALL_ACCESS.getName())))
+		.remote(REMOTE_CLUSTER_NAME, remoteCluster)
+		.roles(LIMITED_ROLE, DLS_ROLE_ROCK, DLS_ROLE_JAZZ, FLS_EXCLUDE_LYRICS_ROLE, FLS_INCLUDE_TITLE_ROLE).authc(AUTHC_HTTPBASIC_INTERNAL)
+		.users(ADMIN_USER, LIMITED_USER, DLS_USER_ROCK, DLS_USER_JAZZ, FLS_INCLUDE_TITLE_USER, FLS_EXCLUDE_LYRICS_USER)
+		.build();
+
+	@ParametersFactory(shuffle = false)
+	public static Iterable<Object[]> parameters() {
+		return List.of(new Object[] { true }, new Object[] { false });
+	}
```

Comment:
Do these create any complexity in debugging that you ran into while authoring?  Just doublechecking if we should update any documentation or consider another design

Reply:
During debugging, the test method is executed twice for each parameter value. If this is inconvenient then it is easy to temporarily remove one parameter value.

I also considered creating an abstract class which contains all test methods and a constructor which accepts the parameter `ccsMinimizeRoundtrips` value. Then two subclasses are used to just invoke the superclass constructor with the proper value of the parameter `ccsMinimizeRoundtrips`. But I think that debugging is easier when the approach with parameterized tests is used.